meetings/working-groups/extensions/metadata-names.md

```diff
@@ -0,0 +1,185 @@
+But this is causing some issues.
+
+The way we produce docIDs for types is to take the name and append the arity suffix.  
+For extensions, we added special handling to use the grouping and marker names. 
+
+Let's consider this example:
+```csharp
+static class E
+{
+  extension<T>(T t)
+  {
+    public void M() { }
+  }
+}
+```
+with corresponding metadata:
+```
+.class E
+{
+  .class '<G>$8048A6C8BE30A622530249B904B537EB'<T0> // grouping type
+  {  
+    .class '<M>$65CB762EDFDF72BBC048551FDEA778ED'<T> // marker type
+    {
+      .. marker method ..
+    }
+    public void M() => throw; // extension member without implementation
+  }
+
+  public static void M(this int i) { } // implementation method
+}
+```
+
+**If we do include an arity suffix** when producing docIDs for extensions, then the docIDs don't match the metadata names.
+
+The docIDs for the example would differ from the names in metadata:
+- extension block: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1.<M>$65CB762EDFDF72BBC048551FDEA778ED"
+- extension member: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1.M"
+
+
+**If we don't include an arity suffix**, then:
+- docIDs produced from VB symbols on extension metadata will diverge from those produce from C# symbols. VB doesn't have the concept of extension, so will have regular handling for types (which include an arity suffix)
+- if someone makes metadata for an extension type using some other tool, and they do include the arity suffix in the metadata names of grouping and marker types, then docIDs won't match metadata names again.
+
+To illustrate the first bullet, the docIDs from C# source or metadata for the example would be:
+- extension block: "E.<G>$8048A6C8BE30A622530249B904B537EB.<M>$65CB762EDFDF72BBC048551FDEA778ED"
+- extension member: "E.<G>$8048A6C8BE30A622530249B904B537EB.M"
+
+But the docIDs from VB metadata would differ from those from C#:
+- extension grouping type: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1"
+- extension marker type: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1.<M>$65CB762EDFDF72BBC048551FDEA778ED"
+- extension member: "E.<G>$8048A6C8BE30A622530249B904B537EB\`1.M"
+
+To illustrate the second bullet, if some other tool cooks up extension metadata including arity suffixes like this:
+```
+.class E
+{
+  .class 'GroupingType`1'<T0> // grouping type
+  {  
+    .class 'MarkerType'<T> // marker type
+    {
+      .. marker method ..
+    }
+    public void M() => throw; // extension member without implementation
+  }
+
+  public static void M(this int i) { } // implementation method
+}
+```
+
+Then the docIDs would not match the metadata names:
+- extension block: "E.GroupingName.MarkerName"
+- extension member: "E.GroupingName.M"
+
+# Proposal 
+
+We're proposing to update the metadata design to include arity suffix in the metadata names for grouping and marker types.  
```

Comment:
>and marker types [](http://example.com/codeflow?start=105&length=16)

Consider dropping marker types from this sentence since no change is proposed for them.